conf/imx_hab4/sign-file.sh

```diff
@@ -0,0 +1,215 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+CST_BINARY="cst"
+CSF_TEMPLATE="u-boot-spl-sign.csf-template"
+KEY_DIR="."
+DCD_CLEAR=0
+DISPLAY_USAGE=0
+SIGN_SPL=0
+SIGN_M4APP=0
+
+function parse_args()
+{
+    while [ ${#} -gt 0 ]
+    do
+        case ${1} in
+        --cst)
+            CST_BINARY=${2}
+            shift
+            shift
+            ;;
+        --csf-template)
+            CSF_TEMPLATE=${2}
+            shift
+            shift
+            ;;
+        --spl)
+            WORK_FILE=${2}
+            SIGN_SPL=1
+            shift
+            shift
+            ;;
+        --m4app)
+            WORK_FILE=${2}
+            SIGN_M4APP=1
+            shift
+            shift
+            ;;
+        --key-dir)
+            KEY_DIR=${2}
+            shift
+            shift
+            ;;
+        --fix-sdp-dcd)
+            DCD_CLEAR=1
+            shift
+            ;;
+        -h)
+            DISPLAY_USAGE=1
+            shift
+            ;;
+        --help)
+            DISPLAY_USAGE=1
+            shift
+            ;;
+        *)
+            shift
+            ;;
+        esac
+    done
+}
+
+parse_args "$@"
+
+if [ "${DISPLAY_USAGE}" == "1" ]; then
+    echo "usage for: ${0}"
+    echo "  --cst: set cst binary path/filename   [default: ${CST_BINARY}]"
+    echo "  --csf-template: set CSF template file [default: ${CSF_TEMPLATE}]"
+    echo "  --spl: SPL binary to sign             [required]"
+    echo "  --m4app: M4 binary to sign            [required]"
+    echo "  --key-dir: location for key files     [default: ${KEY_DIR}]"
+    echo "  --fix-sdp-dcd: turn on clear / restore DCD addr for SPD SPL binary"
+    exit 0
+fi
+
+if [ -z "${WORK_FILE}" ]; then
+    echo "ERROR: must specify either --spl or --m4app"
+    echo 1
+fi
+
+if [ "${SIGN_M4APP}" == "1" ] && [ "${DCD_CLEAR}" == "1" ]; then
+    echo "ERROR: --fix-sdp-dcd is incompatible with --m4app"
+    echo 1
+fi
+
+if [ "${DCD_CLEAR}" == "1" ]; then
+    FIX_SDP_DCD="yes"
+else
+    FIX_SDP_DCD="no"
+fi
+
+echo ""
+echo "SETTINGS FOR  : ${0}"
+echo "--------------:"
+echo "CST BINARY    : ${CST_BINARY}"
+echo "CSF TEMPLATE  : ${CSF_TEMPLATE}"
+echo "BINARY FILE   : ${WORK_FILE}"
+echo "KEYS DIRECTORY: ${KEY_DIR}"
+if [ "${SIGN_SPL}" == "1" ]; then
+    echo "FIX-SDP-DCD   : ${FIX_SDP_DCD}"
+fi
+echo ""
+
+# Transform template -> config
+sed "s/@@KEY_ROOT@@/${KEY_DIR}/g" ${CSF_TEMPLATE} > ${CSF_TEMPLATE}.csf-config
+
+# working file used for signature
+cp ${WORK_FILE} ${WORK_FILE}.mod
+
+# for M4 application: pad binary to 0x1000 alignment
+if [ "${SIGN_M4APP}" == "1" ]; then
+    BINARY_LEN=$(od -An -t x4 -j 0x1024 -N 0x4 ${WORK_FILE}.mod | cut -d' ' -f2)
+    BINARY_LEN=$(printf "%08x" $(((0x${BINARY_LEN} / 0x1000 + 1) * 0x1000)))
+    objcopy -I binary -O binary --pad-to 0x${BINARY_LEN} --gap-fill=0x5A ${WORK_FILE}.mod ${WORK_FILE}.mod
```

Comment:
While objcopy and other tools are commonly available, it is probably a good idea to use 'set -e' as part of this script, in case one of the tools are not available at the host side.

Reply:
Good catch will remove that.